### PR TITLE
Add docs generation tool to goreleaser

### PR DESCRIPTION
**Pull request by feedmeapples**
## What was changed

Added docs generation tool to .goreleaser

## Why?

- simplifies consuming the tool from documentation repo, including fetching the tool for specific Temporal CLI version

## Checklist

1. Closes 

2. How was this tested:

ran .goreleaser locally

3. Any docs updates needed?

### CODE REVIEW

**Review by DJSanti — APPROVED** [review]
Thanks! I approve, but make sure someone else beside me does (and that it works for them as well)

File: .goreleaser.yml [hunk 1]
```diff
@@ -25,6 +26,19 @@ builds:
       - amd64
       - arm64
 
+  - id: temporal-doc-gen
```
Comment (bergundy) [inline .goreleaser.yml:29]: Can we make sure this isn't added to the release archive that users consume please?

File: Makefile [hunk 2]
```diff
@@ -24,10 +24,16 @@ PINNED_DEPENDENCIES := \
 
 ##### Build #####
 
-build:
-	@printf $(COLOR) "Build Temporal CLI with OS: $(GOOS), ARCH: $(GOARCH)..."
+build: build-cli build-doc-gen
+
+build-cli:
```
Comment (bergundy) [inline Makefile:29]: Why do we even need a makefile in this repo? Doesn't seem like there's that much value.

Comment (feedmeapples) [reply Makefile:29]: done

Comment (feedmeapples) [reply Makefile]: sure removed. I guess mostly just to have makefile consistent with the tools in the repo

File: .goreleaser.yml (hunk 1 above)
Comment (feedmeapples) [reply .goreleaser.yml:36]: done

**Review by bergundy — APPROVED** [review]

**Pull request merged**